lib/pure/ropes.nim

```diff
@@ -171,28 +220,73 @@ proc `&`*(a, b: Rope): Rope {.rtl, extern: "nroConcRopeRope".} =
     result.right = b
 
 proc `&`*(a: Rope, b: string): Rope {.rtl, extern: "nroConcRopeStr".} =
-  ## the concatenation operator for ropes.
+  ## The concatenation operator for ropes.
+  runnableExamples:
+    var
+      r1 = rope("Hello, ")
+      r2 = "Nim!"
+    
+    let r = r1 & r2
+    doAssert $r == "Hello, Nim!"
   result = a & rope(b)
 
 proc `&`*(a: string, b: Rope): Rope {.rtl, extern: "nroConcStrRope".} =
-  ## the concatenation operator for ropes.
+  ## The concatenation operator for ropes.
+  runnableExamples:
+    var
+      r1 = "Hello, "
+      r2 = rope("Nim!")
+    
+    let r = r1 & r2
+    doAssert $r == "Hello, Nim!"
   result = rope(a) & b
 
 proc `&`*(a: openArray[Rope]): Rope {.rtl, extern: "nroConcOpenArray".} =
-  ## the concatenation operator for an openarray of ropes.
-  for i in countup(0, high(a)): result = result & a[i]
+  ## The concatenation operator for an openarray of ropes.
+  runnableExamples:
+    let s = @[rope("Hello, "), rope("Nim"), rope("!")]
+    let r = &s
+    doAssert $r == "Hello, Nim!"
+  for item in a: result = result & item
```

Comment:
small improvement here
`for i in countup(0, high(a)): result = result & a[i]` > `for item in a: result = result & item`